src/components/QuestionCard.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+import OpenGraphMeta from "./OpenGraphMeta"
+import styled from "styled-components";
+import { useHistory } from "react-router-dom";
+
+
+const getQuestionById = (questionId) => {
+  /* TODO: Get question data via question ID
+     Expect return result: dict() */
+  const result = {
+    dateCreated: "July 1, 2020",
+    text: "Is it true that everything happens for a reason?",
+    newsUrl: "https://www.notion.so/Contentforslides28bb12e73c124253b1513f5aa4cb1ec3",
+    newsTitle: "Is it true that everything happens for a reason? Is it true that everything happens for a reason? Is it true that everything happens for a reason? Is it true that everything happens for a reason?",
+    newsImage: "https://media.proprofs.com/images/QM/user_images/2503852/1568196428.jpg"
+  }
+  return result
+}
+
+const getAnswerByQuestionId = (questionId) => {
+  /* TODO: Get question data via question ID
+     Expect return result: dict() */
+  const result = {
+    totalCount: 42,
+    numberOfTrue: 32,
+    numberOfFalse: 10
+  }
+  return result
+}
+
+const handleClick = (history, question_id) => {
+  history.push("/question/" + question_id);
+}
+
+export default function QuestionCard(props) {
+  const [question, setQuestion] = useState({ dateCreated: "TBC", text: "", newsUrl: "", newsTitle: "", newsImage: "" })
```

Comment:
None of this is going to be necessary. It's all going to come straight from the API and there will be no need to `setQuestion`.

Reply:
Relay will inject all the data directly into your component.

Reply:
You still variable to hold those value is't it?

Reply:
It will be component props